### PR TITLE
ci: move to github actions

### PR DESCRIPTION
**Pull request by irgendwr**
move build process from travis to github actions

### CODE REVIEW

**Review by Multivit4min — APPROVED** [review]

**Review by mxschmitt — APPROVED** [review]
what about the NPM deployment?

**Comment by irgendwr** [comment]
> what about the NPM deployment?

good catch, I'll add that too :+1:

File: .github/workflows/deployment.yml [hunk 1]
```diff
@@ -28,3 +29,14 @@ jobs:
...
         BRANCH: gh-pages
         FOLDER: documentation
+    - name: Publish to npm
+      run: npm publish --access public
+      env:
+        NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+    - uses: actions/setup-node@v1
+      with:
+        registry-url: 'https://npm.pkg.github.com'
+    - name: Publish to GitHub Packages
```
Comment (mxschmitt) [inline .github/workflows/deployment.yml:42]: I would not publish it to GitHub packages too

File: .github/workflows/deployment.yml [hunk 2]
```diff
@@ -28,3 +29,14 @@ jobs:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         BRANCH: gh-pages
         FOLDER: documentation
+    - name: Publish to npm
```
Comment (mxschmitt) [inline .github/workflows/deployment.yml:32]: You only want to execute that usually when you create a release.

Or at least only on master if it won't error when the version already exists.

**Review by mxschmitt — CHANGES_REQUESTED** [review]

File: .github/workflows/deployment.yml (hunk 1 above)
Comment (irgendwr) [reply .github/workflows/deployment.yml:42]: I'll just give it a try anyway. The neat thing is that this does not need an npm token, so things are more self-contained.

**Review by mxschmitt — APPROVED** [review]

**Pull request merged**